### PR TITLE
Move initialization functions to dedicated module (ref #137)

### PR DESCRIPTION
**Pull request by leplatrem**
r?

-----------
I chose not to expose the configuration of steps in the documentation yet. Prove me wrong :)

### CODE REVIEW

File: cliquet/__init__.py [hunk 1]
```diff
@@ -122,272 +101,31 @@ def load_default_settings(config):
...
-
-    id_generator = config.maybe_dotted(settings['cliquet.id_generator'])
-    config.registry.id_generator = id_generator()
-
-    set_auth(config)
-    attach_http_objects(config)
+    # Monkey Patch Cornice Service to setup the global CORS configuration.
+    # XXX: Refactor @crud decorator and inherit Service instead.
+    cors_origins = settings['cliquet.cors_origins']
+    Service.cors_origins = (cors_origins,)
```
Comment (leplatrem) [inline cliquet/__init__.py]: should probably be ``aslist(cors_origins)``

Comment (ametaireau) [inline cliquet/__init__.py]: Well, having something imutable here is something I think we should keep. `tuple(aslist(cors_origins))` then :)

File: cliquet/__init__.py [hunk 2]
```diff
@@ -122,272 +101,31 @@ def load_default_settings(config):
...
+    Service.default_cors_headers = ('Backoff', 'Retry-After', 'Alert')
 
-    # Handle StatsD
-    handle_statsd(config)
+    # Setup components.
+    for step in aslist(settings['cliquet.initialization_steps']):
+        setup_func = config.maybe_dotted(step)
```
Comment (ametaireau) [inline cliquet/__init__.py]: setup doesn't seem right here, I would use `step_location` and `step`, or maybe `step_callable`.

**Comment by ametaireau** [comment]
Not documenting is IMO an issue here. We obviously don't want to have this show up in the docs as something everyone needs to know, but that's a pretty powerful thing we should expose.

Also, this approach doesn't allow to easily manipulate the steps, like we would do in python for instance (add, remove). I'm not sure what we should do about that (add more settings with a `remove` or `add` prefix?) but if we consider explicit is better than implicit then that should work out.

Also, once we introduce this, we will probably have problems when we want to rename / move our initializers somewhere else, because users will be relying on their location.

**Pull request merged**